pkg/engine/armvariables.go

```diff
@@ -91,6 +91,7 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		"maximumLoadBalancerRuleCount":  maxLoadBalancerCount,
 		"masterFqdnPrefix":              "[tolower(parameters('masterEndpointDNSNamePrefix'))]",
 		"apiVersionCompute":             "2018-10-01",
+		"apiVersionDeployments":         "2018-06-01",
```

Comment:
this essentially reverts #578 for extensions

Reply:
how did you determine that version? I had a hard time finding out which docs to look at for the deployment versions.

Reply:
the docs point to `2018-05-01` https://docs.microsoft.com/en-us/rest/api/resources/deployments but I was running into errors with that version so I used the one that was known working before https://github.com/Azure/aks-engine/pull/578/files#diff-aa4ff2b7c6ff5dfd7e2014283a7aa696L99.